### PR TITLE
Throw TypeError instead of SyntaxError when parsing a URL fails

### LINKED ISSUE #268: Throw "TypeError" instead of "Syntax" when parsing a URL fails
>2. Let urlRecord be the result of [parsing ](https://url.spec.whatwg.org/#concept-url-parser)record's data.
>3. If urlRecord is failure, throw a "SyntaxError" DOMException and abort these steps.

>4. Let urlRecord be the result of [parsing ](https://url.spec.whatwg.org/#concept-url-parser)id.
>5. If urlRecord is failure, throw a "SyntaxError" DOMException and abort these steps.

These would be better to throw "TypeError". 

@kenchris @marcoscaceres

### PR DESCRIPTION
**Pull request by Honry**
Fixed #268

***
<a href="https://pr-preview.s3.amazonaws.com/Honry/web-nfc/pull/270.html" title="Last updated on Aug 5, 2019, 7:08 AM UTC (67c8d06)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/web-nfc/270/8220627...Honry:67c8d06.html" title="Last updated on Aug 5, 2019, 7:08 AM UTC (67c8d06)">Diff</a>

### CODE REVIEW

**Comment by Honry** [comment]
@kenchris, @marcoscaceres, PTAL. thanks~!

File: index.html [hunk 1]
```diff
@@ -2543,7 +2543,7 @@ <h3><dfn>Writing or pushing content</dfn></h3>
           </li>
           <li>
             If |urlRecord| is failure, [= exception/throw =] a
-            {{"SyntaxError"}} {{DOMException}} and abort these steps.
+            {{TypeError}} {{DOMException}} and abort these steps.
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
            {{TypeError}} and abort these steps.
```

File: index.html [hunk 2]
```diff
@@ -2724,7 +2724,7 @@ <h3><dfn>Writing or pushing content</dfn></h3>
...
```
Comment (marcoscaceres) [inline index.html]:
```suggestion
            {{TypeError}} and abort these steps.
```

File: index.html (hunk 1 above)
Comment (marcoscaceres) [reply index.html]: "and abort these steps" is not technically needed.

Comment (marcoscaceres) [reply index.html]: Remember that DOMExceptions are named, like "AbortError"... while TypeError is an interface :)

Comment (Honry) [reply index.html]: So when should it be there? "and abort these steps" are everywhere in the steps that throws Exceptions.

Comment (marcoscaceres) [reply index.html]: Yeah, you can probably remove them all. Throwing implies halting (like in normal programming) so it's not needed... however, it's a matter of style, so @kenchris should decide as primary editor.

Comment (Honry) [reply index.html]: Thanks, ping @kenchris.

**Review by kenchris — APPROVED** [review]

**Pull request merged**